etna/packages/etna-js/upload/workers/uploader.jsx

```diff
@@ -174,14 +176,18 @@ export default (uploader) => {
 
     remove: (upload) => uploader.dispatch({ type: 'REMOVE_UPLOAD', upload }),
 
-    timeout: () => {
-      ++uploader.timeouts;
-      if (uploader.timeouts == uploader.maxTimeouts) {
-        uploader.reset();
+    timeout: (upload) => {
+      const { url } = upload;
+      const { timeouts, maxTimeouts } = uploader;
+
+      timeouts[url] = timeouts[url] || 0;
+
+      if (++timeouts[url] >= maxTimeouts) {
         uploader.dispatch({ type: 'UPLOAD_TIMEOUT', upload });
-        return false;
```

Comment:
This boolean was incorrect; timeout should return true when there is a timeout, and false when there is not.  Before, it would stall immediately after any one failure.